ios/Classes/Plugin/PluginController.swift

```diff
@@ -208,6 +208,19 @@ final class PluginController {
         let timeout = args.timeoutInMs > 0 ? TimeInterval(args.timeoutInMs) / 1000 : nil
 
         completion(.success(nil))
+        
+        guard let sink = connectedDeviceSink
+        else {
+            print("Warning! No event channel set up to report a connection update")
+            return
+        }
```

Comment:
With your current implementation this is not "just" a warning. When you're returning from here you're canceling the connection request.

One way around this is to rearrange the code in the following manner:

```swift
if let sink = connectedDeviceSink {
    let message = DeviceInfo.with {
        $0.id = args.deviceID
        $0.connectionState = encode(.connecting)
    }

    sink.add(.success(message))
} else {
    print("Warning! No event channel set up to report a connection update")
}
```

Reply:
@werediver  thanks for the lesson I added a new fix with your suggestion but the problem I have there is that I need to keep put back the guard method in the do block since sink is then out of scope of that block.

I considered an alternative to make the variable sink optional but that goes a bit against the purpose since `connectedDeviceSink` is already optional

Reply:
At least this way the old behavior is preserved as much as possible. I'm not implying it cannot or shouldn't be changed, but that should be though through.

I hope we're reliably getting the sink here 🙄